File: headless_solutions/base_7/solution.cpp
#include <string>
#include <algorithm> // For std::reverse

class Solution {
public:
    std::string convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }

        bool is_negative = num < 0;
        // Work with absolute value for conversion
        int abs_num = std::abs(num); 
        
        std::string result_digits = ""; // Use string to build digits

        while (abs_num > 0) {
            int remainder = abs_num % 7;
            // Convert remainder to char and append.
            // '0' + remainder converts int 0-6 to char '0'-'6'
            result_digits += (char)('0' + remainder); 
            abs_num /= 7;
        }

        // The digits are collected in reverse order, so reverse the string
        std::reverse(result_digits.begin(), result_digits.end());

        if (is_negative) {
            return "-" + result_digits;
        } else {
            return result_digits;
        }
    }
};

File: headless_solutions/base_7/solution.js
/**
 * @param {number} num
 * @return {string}
 */
var convertToBase7 = function(num) {
    if (num === 0) {
        return "0";
    }

    let isNegative = num < 0;
    // Work with absolute value for conversion
    let absNum = Math.abs(num);
    
    let digits = []; // Use an array to store digits

    while (absNum > 0) {
        let remainder = absNum % 7;
        digits.push(remainder); // Append the digit (as number, will be converted to string by join)
        absNum = Math.floor(absNum / 7); // Integer division
    }

    // The digits are collected in reverse order, so reverse the array
    digits.reverse(); 

    // Join the array of digits into a single string
    let resultStr = digits.join("");

    if (isNegative) {
        return "-" + resultStr;
    } else {
        return resultStr;
    }
};
